libc/src/__support/float_to_string.h

```diff
@@ -152,15 +157,17 @@ LIBC_INLINE constexpr uint32_t ceil_log10_pow2(const uint32_t e) {
   return log10_pow2(e) + 1;
 }
 
+LIBC_INLINE constexpr uint32_t div_ceil(const uint32_t num,
+                                        const uint32_t denom) {
+  return (num + (denom - 1)) / denom;
```

Comment:
Can you double check if  this division is optimized to multiplication or shifts in the generated codes since `BLOCK_SIZE` is constant?

Reply:
It definitely is, look at this godbolt link: https://godbolt.org/z/czh87M61b

Reply:
`const` quantifiers in the function signatures don't do anything with value types.  You should remove them to reduce confusion with the function signatures.  Same with the function below.